src/BinSkim.Sdk/CompilerDataLogger.cs

```diff
@@ -77,9 +81,15 @@ public static void Initialize(string instrumentationKey)
         {
             if (s_telemetryConfiguration == null && s_telemetryClient == null)
             {
-                s_sessionId = Guid.NewGuid().ToString();
-                s_telemetryConfiguration = new TelemetryConfiguration(instrumentationKey);
-                s_telemetryClient = new TelemetryClient(s_telemetryConfiguration);
+                lock (s_syncRoot)
```

Comment:
when we changed from single thread to multi thread, we started to create multiple configurations/clients/sessions.

With that in mind, we must synchronize the threads to prevent this behavior.